### DISCUSSION TITLE
Add agglemorative clustering with minmax

### DISCUSSION THREAD

**Discussion by NimaSarajpoor** [rasbt/mlxtend#960] [Ideas]
So, this is the idea that I have had for a while and I also mentioned it in scipy and scikit-learn-extra repos but I think they are a little bit busy and I was also busy at that time. 

The idea is to create a module for bottom-up agglemorative clustering with linkage minmax, introduced in a paper by Dr. Tibshirani (see [paper](https://www.tandfonline.com/doi/abs/10.1198/jasa.2011.tm10183)). I implemented it myself but it was not clean and I will probably need help throughout the process to make it efficient and clean.

The advantage of this linkage is that it can be used with any distance measure AND provide a meaningful centroid. Although we can use the linkage average / single/ complete with different distance measures, no centroid is defined for such clusters. However, the min-max linkage is fast, it can be used with different distances (e.g. DTW), and it provides a well-defined centroid which is important if someone is looking for a prototype.

I took a look at mlxtend `cluster` and I noticed it only supports K-Means. `scikit-learn-extra` provides K-Medoids, and `scikit-learn` provides agglemorative clustering for other linkages. However, I haven't found any implementation of min-max.

I think min-max is something cool that can be implemented and I think most researchers are not aware of this linkage.

---

If you think it is a good idea, I can create an issue and start working on it :)

**Comment by rasbt** [comment]
That's interesting, thanks for suggesting. Personally, I haven't used or read about linkage with min-max. I wonder, in this case, if this shouldn't be added to the core scikit-learn library as an additional "linkage" option? I think it might be beneficial this way, because their general implementation is very efficient and fast.

**Reply by NimaSarajpoor** [reply]
> . I wonder, in this case, if this shouldn't be added to the core scikit-learn library as an additional "linkage" option?

Yeah...that makes more sense as they already have support for other linkages. I think I should go back and re-visit their code :) Thanks for the quick response.
